manager/src/main/java/io/gingersnapproject/k8s/CacheRuleInformer.java

```diff
@@ -0,0 +1,357 @@
+
+      ConfigMapEventHandler lazyHandler = new ConfigMapEventHandler(rulesConfig,
+      LazyCacheRuleSpec.getDefaultInstance());
+      ConfigMapEventHandler eagerHandler = new ConfigMapEventHandler(rulesConfig,
+      EagerCacheRuleSpec.getDefaultInstance());
+
+      // Start informers
+      lazyInformer = lazyCm.inform(lazyHandler, RESYNC_PERIOD);
+      lazyInformer.start();
+      eagerInformer = eagerCm.inform(eagerHandler, RESYNC_PERIOD);
+      eagerInformer.start();
+   }
+
+   public void stop(@Observes ShutdownEvent ignore) {
+      if (lazyInformer != null) {
+         log.info("Shutdown lazyInformer");
+         lazyInformer.close();
+      }
+
+      if (eagerInformer != null) {
+         log.info("Shutdown eagerInformer");
+         eagerInformer.close();
+      }
+   }
+
+}
+
+class ConfigMapEventHandler implements ResourceEventHandler<ConfigMap> {
+   private static final Logger log = LoggerFactory.getLogger(ConfigMapEventHandler.class);
+   private Configuration lrc;
+   private GeneratedMessageV3 ruleTemplate;
+
+   private interface SendEventFunc {
+      void sendEvent(String name, Message rule);
+   }
+
+   public ConfigMapEventHandler(Configuration lrc, GeneratedMessageV3 rule) {
+      this.lrc = lrc;
+      this.ruleTemplate = rule;
+   }
+
+   private void processConfigMapAndSend(Set<Map.Entry<String, String>> entries, SendEventFunc f) {
+      for (Entry<String, String> entry : entries) {
+         var eRuleBuilder = ruleTemplate.newBuilderForType();
+         try {
+            JsonFormat.parser().ignoringUnknownFields().merge(entry.getValue(), eRuleBuilder);
+         } catch (InvalidProtocolBufferException e) {
+            log.error("Cannot parse eager rule with name {}", entry.getKey(), e);
+         }
+         f.sendEvent(entry.getKey(), eRuleBuilder.build());
+      }
+   }
+
+   @Override
+   public void onAdd(ConfigMap obj) {
+      log.debug("calling onAdd for ConfigMap {}",obj.getMetadata().getName());
+      processConfigMapAndSend(obj.getData().entrySet(), (name, rule) -> {
+         log.debug("calling Configuration.addRule({},{}", name, rule.toString());
+         lrc.addRule(name, CacheRuleSpecAdapterFactory.newAdapterRule(rule));
+      });
+   }
+
+   @Override
+   public void onUpdate(ConfigMap oldObj, ConfigMap newObj) {
+      Map<String, String> oldMap = oldObj.getData();
+      Map<String, String> newMap = newObj.getData();
+      var olds = oldMap.entrySet();
+      var news = newMap.entrySet();
+
+      checkNoUpdatesOrThrow(oldMap, newMap, olds, news);
+
+      var added = new HashSet<Map.Entry<String, String>>(news);
+      // Get added keys. From new set remove
+      // - keys already present in old set
+      added.removeIf(arg0 -> {
+         return oldMap.containsKey(arg0.getKey());
+      });
+
+      var removed = new HashSet<Map.Entry<String, String>>(olds);
+      // Get removed keys. From old set remove
+      // - keys still present in new set
+      removed.removeIf((arg0 -> {
```

Comment:
Can be simplified to a one-liner `added.removeIf(arg0 -> oldMap.containsKey(arg0.getKey()))`